src/providers/trakt.ts

```diff
@@ -0,0 +1,67 @@
+import { OAuthConfig, OAuthUserConfig } from "."
+
+export interface TraktUser {
+  username: string
+  private: boolean
+  name: string
+  vip: boolean
+  vip_ep: boolean
+  ids: { slug: string }
+  joined_at: string
+  location: string | null
+  about: string | null
+  gender: string | null
+  age: number | null
+  images: { avatar: { full: string } }
+}
+
+export default function Trakt<
+  P extends Record<string, any> = TraktUser
+>(options: OAuthUserConfig<P>): OAuthConfig<P> {
+  return {
+    id: "trakt",
+    name: "Trakt",
+    type: "oauth",
+    authorization: {
+      url: "https://trakt.tv/oauth/authorize",
```

Comment:
>Note: You should use the normal https://trakt.tv/ hostname when creating this URL and not the API URL.

The trakt [documentation](https://trakt.docs.apiary.io/#reference/authentication-oauth/authorize/authorize-application) recommends not using the api URL for authentication. So I'm not too sure about this.

Reply:
Confused myself:

![image](https://user-images.githubusercontent.com/18369201/152361264-ba140aa0-4af9-45a3-adc8-8b308b750f1c.png)


But since both works, I won't pick on this more :sweat_smile: if it's a problem in the future, we can revisit. Thanks!

Reply:
And that's what caused the first auth issue as well 😆